cmd/registry-replacer/main.go

```diff
@@ -14,24 +16,73 @@ import (
 	"github.com/sirupsen/logrus"
 	utilerrors "k8s.io/apimachinery/pkg/util/errors"
 	"k8s.io/apimachinery/pkg/util/sets"
+	"k8s.io/test-infra/experiment/autobumper/bumper"
+	"k8s.io/test-infra/prow/config/secret"
+	"k8s.io/test-infra/prow/flagutil"
+	"k8s.io/test-infra/prow/github"
+	"k8s.io/test-infra/prow/labels"
 	"sigs.k8s.io/yaml"
 
 	"github.com/openshift/ci-tools/pkg/api"
 	"github.com/openshift/ci-tools/pkg/config"
 )
 
-func main() {
-	configDir := flag.String("config-dir", "", "The directory with the ci-operator configs")
+type options struct {
+	configDir      string
+	createPR       bool
+	githubUserName string
+	selfApprove    bool
+	flagutil.GitHubOptions
+}
+
+func gatherOptions() (*options, error) {
+	o := &options{}
+	flag.StringVar(&o.configDir, "config-dir", "", "The directory with the ci-operator configs")
+	flag.BoolVar(&o.createPR, "create-pr", false, "If the tool should automatically create a PR. Requires --token-file")
+	flag.StringVar(&o.githubUserName, "github-user-name", "openshift-bot", "Name of the github user. Required when --create-pr is set. Does nothing otherwise")
+	flag.BoolVar(&o.selfApprove, "self-approve", false, "If the bot should self-approve its PR.")
```

Comment:
Do we want a common options struct to hold these and wrap the interaction with the library?

Reply:
not really worth it at this point, this is about to options, a boolean and a string, I don't think we gain much from sharing that